### PR TITLE
[SPARK-32845][SS][TESTS] Add sinkParameter to check sink options robustly in DataStreamReaderWriterSuite

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to add `sinkParameter`  to check sink options robustly and independently in DataStreamReaderWriterSuite

### Why are the changes needed?

`LastOptions.parameters` is designed to catch three cases: `sourceSchema`, `createSource`, `createSink`. However, `StreamQuery.stop` invokes `queryExecutionThread.join`, `runStream`, `createSource` immediately and reset the stored options by `createSink`.

To catch `createSink` options, currently, the test suite is trying a workaround pattern. However, we observed a flakiness in this pattern sometimes. If we split `createSink` option separately, we don't need this workaround and can eliminate this flakiness.

```scala
val query = df.writeStream.
   ...
   .start()
assert(LastOptions.paramters(..))
query.stop()
```

### Does this PR introduce _any_ user-facing change?

No. This is a test-only change.

### How was this patch tested?

Pass the newly updated test case.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala [hunk 1]
```diff
@@ -43,11 +43,13 @@ object LastOptions {
   var mockStreamSourceProvider = mock(classOf[StreamSourceProvider])
   var mockStreamSinkProvider = mock(classOf[StreamSinkProvider])
   var parameters: Map[String, String] = null
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala:45]: I keep this name to reduce the patch size.

**Comment by dongjoon-hyun** [comment]
@cloud-fan , could you review this?

**Comment by dongjoon-hyun** [comment]
Also, cc @viirya and @HeartSaVioR

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala (hunk 1 above)
Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala:45]: How about add a comment?

**Review by viirya — COMMENTED** [review]
Looks okay. I'm just curious why StreamQuery.stop invokes createSource again?

**Comment by dongjoon-hyun** [comment]
It's because `stop` calls `queryExecutionThread.join` immediately and notifies the thread, the the thread runs `runStream` -> `override lazy val logicalPlan: LogicalPlan` -> `dataSourceV1.createSource(metadataPath)`.

**Comment by dongjoon-hyun** [comment]
Merged to master/3.0/2.4.

**Comment by cloud-fan** [comment]
@dongjoon-hyun thanks for fixing it!

**Comment by dongjoon-hyun** [comment]
Thanks! :)

**Pull request closed without merging**